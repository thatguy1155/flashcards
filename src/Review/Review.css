.review-page{
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 90vh;
}
.nav-buttons{
  width:100%;
  display: flex;
  justify-content:space-around;
}