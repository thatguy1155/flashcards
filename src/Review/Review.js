import React from 'react';
import Card from './components/card/card';
import './Review.css';
import { FaArrowLeft, FaArrowRight } from 'react-icons/fa';

export default function Review() {
  return (
    <div className="review-page">
      <div className="nav-buttons">
        <FaArrowLeft />
        <FaArrowRight />
      </div>
      <Card english="hello" korean="안녕" language="eng" />
    </div>
  );
}
