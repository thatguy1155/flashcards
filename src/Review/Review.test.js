import TestRenderer from 'react-test-renderer';
import Card from './components/card/card';
import {
  act, render, screen, fireEvent, within,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Review from './Review';

describe('Header', () => {
  test('non-shallow render', () => {
    // eslint-disable-next-line new-cap
    const element = new TestRenderer.create(
      <Review />,
    );
    expect(element.toJSON()).toMatchSnapshot();
  });

  it('displays and English or Korean word', () => {
    const card = render(<Card english="hey" korean="안녕" language="eng" />);

    const display = card.getByTestId('display');
    const checkButton = card.getByTestId('check-button');
    const { getByText } = within(display);

    expect(getByText('hey')).toBeInTheDocument();
    act(() => {
      userEvent.click(checkButton);
    });
    expect(getByText('안녕')).toBeInTheDocument();
  });
});
