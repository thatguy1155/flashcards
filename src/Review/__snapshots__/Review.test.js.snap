// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header non-shallow render 1`] = `
<div
  className="review-page"
>
  <div
    className="nav-buttons"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
      />
    </svg>
  </div>
  <div
    className="card"
  >
    <h1
      className="displayed-text"
      data-testid="display"
    >
      hello
    </h1>
    <button
      data-testid="check-button"
      onClick={[Function]}
      type="button"
    >
      check
    </button>
  </div>
</div>
`;
